src/checks/incorrect_delimiter.rs

```diff
@@ -24,7 +24,13 @@ impl Default for IncorrectDelimiterChecker<'_> {
 impl Check for IncorrectDelimiterChecker<'_> {
     fn run(&mut self, line: &LineEntry) -> Option<Warning> {
         let key = line.get_key()?;
-        if key.trim().chars().any(|c| !c.is_alphanumeric() && c != '_') {
+
+        // delimiters occur /between/ characters, not as the initial character, so we should only check those cases
+        if key
+            .trim()
+            .starts_with(|c: char| c.is_alphabetic() || c == '_')
+            && key.trim().chars().any(|c| !c.is_alphanumeric() && c != '_')
```

Comment:
What do you think about that case?
```
*MY-VARIABLE=test
```
it's has incorrect delimiter, but starts with `*` therefore warnings don't will be create.

Reply:
Thanks for catching that. I have a fix in mind: I think we should remove **all** bad characters from the beginning of the key before we perform the IncorrectDelimiter check, does that sound reasonable?

Similarly, for the LeadingCharacterFixer, the code I have written will not turn ```***KEY=VALUE``` into ```KEY=VALUE```, it will only remove one leading character and turn it into ```**KEY=VALUE```, so I think I need to fix that that before this should be merged. I will work on that a little later today, if all of that sounds reasonable.

Reply:
I added a helper function to ```common.rs``` called ```remove_all_invalid_leading_chars``` to do what I described above, and used it both in IncorrectDelimiterChecker and LeadingCharacterFixer. I added some additional tests that should have caught these as well.

Reply:
Sorry, I have been busy last couple days.
I'll look the changes soon. 

@dotenv-linter/core what do you think? This may affect #246

Reply:
I think that these changes in the `IncorrectDelimiterChecker` should be in another PR.